connectors/http/polling/src/main/java/io/camunda/connector/http/polling/service/HttpPollingSharedExecutorService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.http.polling.service;
+
+import io.camunda.connector.http.polling.task.AbstractScheduledTask;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpPollingSharedExecutorService {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(HttpPollingSharedExecutorService.class);
+
+  private static final HttpPollingSharedExecutorService INSTANCE =
+      new HttpPollingSharedExecutorService();
+  private static final int THREAD_POOL_SIZE = 5;
+
+  private final ScheduledExecutorService executorService;
+  private final Map<String, Future<?>> tasks;
+
+  private HttpPollingSharedExecutorService() {
+    this.executorService = Executors.newScheduledThreadPool(THREAD_POOL_SIZE);
+    this.tasks = new ConcurrentHashMap<>();
+    addShutdownHook();
+  }
+
+  public static HttpPollingSharedExecutorService getInstance() {
+    return INSTANCE;
+  }
+
+  public void scheduleWithFixedDelay(
+      AbstractScheduledTask task, long initialDelay, long delay, TimeUnit unit) {
+    ScheduledFuture<?> future =
+        executorService.scheduleWithFixedDelay(task, initialDelay, delay, unit);
+    tasks.put(task.getId(), future);
+  }
```

Comment:
Please use `computeIfAbsent` and schedule only if not present. Please rewrite the code to make it thread-safe:
```
scheduleWithFixedDelay(taskId,  long initialDelay, long delay, TimeUnit unit, Function<String, AbstractScheduledTask> taskCreator) {
tasks.computeIfAbsent("taskId", () -> taskCreator) 
}
```